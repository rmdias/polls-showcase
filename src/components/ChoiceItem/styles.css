.poll__choice_item {
  cursor: pointer;
  margin-bottom: 5px;
  display: flex;
  justify-content: space-between;
  transition: all .25s ease;
}

.poll__choice_selected {
  box-shadow: 0 0 4px 3px #00bbce;
}

@media only screen and (max-width: 768px) {
  .poll__choice_item {
    flex-wrap: wrap;
  }

  .poll__choice_percentage {
    margin-top: 10px;
    margin-left: 15%;
  }
}

