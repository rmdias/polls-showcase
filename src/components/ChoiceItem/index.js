import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

import { Area } from 'luna-ui-lib'

import PercentageBar from 'components/PercentageBar'

import './styles.css'

export default class ChoiceItem extends PureComponent {
  static propTypes = {
    active: PropTypes.bool,
    choice: PropTypes.object,
    totalvotes: PropTypes.number,
    biggernumber: PropTypes.number
  }

  render() {
    const values = [
      { color: '#b2e2bf', amount: this.props.choice.votes },
      { color: '#c7c7c7', amount: this.props.biggernumber }
    ]
    const active = this.props.active && 'poll__choice_selected'

    return (
      <div onClick={this.props.onClick}>
        <Area className={`poll__choice_item ${active}`}>
          <div><strong>{ this.props.choice.choice }</strong></div>
          <div>{ this.props.choice.votes } votes</div>
          <div>{ Math.floor(((this.props.choice.votes / this.props.totalvotes) || 0) * 100) }%</div>
          <div className="poll__choice_percentage"><PercentageBar values={values} width={200}/></div>
        </Area>
      </div>
    )
  }
}
