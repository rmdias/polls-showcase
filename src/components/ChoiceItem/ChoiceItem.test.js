import React from 'react'
import renderer from 'react-test-renderer'

import ChoiceItem from './index'
import ChoiceMock from './__mocks__/__mock__choice'

import { Theme } from 'luna-ui-lib'

describe('<ChoiceItem />', () => {
  it('toMatchSnapshot: Load <ChoiceItem /> component', () => {
    const component = (
      <Theme>
        <ChoiceItem
          choice={ChoiceMock}
          active={true}
          totalvotes={100}
          biggernumber={200}
          onClick={() => this.handleVotingSelection(choice.url)}
        />
      </Theme>
    )

    const tree = renderer
      .create(component)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
