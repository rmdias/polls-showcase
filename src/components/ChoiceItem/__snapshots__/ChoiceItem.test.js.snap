// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChoiceItem /> toMatchSnapshot: Load <ChoiceItem /> component 1`] = `
<div>
  <div
    onClick={[Function]}
  >
    <div
      className="poll__choice_item poll__choice_selected sc-kpOJdX ddRiWz"
    >
      <div>
        <strong>
          ðŸšŽ
        </strong>
      </div>
      <div>
        150
         votes
      </div>
      <div>
        150
        %
      </div>
      <div
        className="poll__choice_percentage"
      >
        <div
          style={
            Object {
              "display": "flex",
              "width": 200,
            }
          }
        >
          <div
            style={
              Object {
                "backgroundColor": "#b2e2bf",
                "fontSize": undefined,
                "height": "16px",
                "padding": "1px 2px",
                "width": 85.71428571428571,
              }
            }
          />
          <div
            style={
              Object {
                "backgroundColor": "#c7c7c7",
                "fontSize": undefined,
                "height": "16px",
                "padding": "1px 2px",
                "width": 114.2857142857143,
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
